docs/implementing-real-time-reporting.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Implementing real-time reporting
+custom_edit_url: https://github.com/stryker-mutator/mutation-testing-elements/edit/master/docs/implementing-real-time-reporting.md
+---
+
+Real-time reporting was recently implemented in Stryker. This page is created to help other mutation testing frameworks to make use the same functionality.
+
+## How does it work?
+
+Currently, Stryker provides the report with updates using [Server Sent Events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events). This was chosen over WebSockets because the dataflow should only be from a mutation framework to the report. The report should not have to send anything to the mutation framework.
+
+## Implementation details
+
+### Preparing the report
+
+First things first, the mutation testing framework should be able to generate an empty report. This report has all information necessary except the tested state of a mutant. These mutants must have the state `Pending`.
```

Comment:
Is this not optional? I don't see why but if a project doesn't want to generate this initial state the report still works right?

Reply:
I'm not sure how one could do this without the initial state...

Reply:
Doesn't the mutant update contain pretty much everything one needs to build the report?

The only initial state absolutely necessary is the files, right?

Reply:
No, it is not possible to do this without including the mutants. When a `mutant-tested` event is sent, it will use the `id` property to find the already present mutant and update its properties. We cannot add new mutants to the files when sending events.

Reply:
Right, perhaps we could note that as a possible future feature?

Reply:
Good idea :+1: